### ISSUE TITLE
ffi pattern matching for pointers in args is broken

### ISSUE THREAD

**Issue by KaruroChori** [saghul/txiki.js#651]
At the moment
```
int Fl_Window_x(Fl_Window *);
```
will generate as AST of its argument:
```json
            {
                "kind": "vardef",
                "type": {
                    "kind": "type",
                    "typeModifiers": [],
                    "name": "Fl_Window",
                    "ptr": 0
                },
                "name": "*"
            }
```
in place of 
```json
            {
                "kind": "vardef",
                "type": {
                    "kind": "type",
                    "typeModifiers": [],
                    "name": "Fl_Window",
                    "ptr": 1
                },
                "name": ""
            }
```

**Comment by saghul** [comment]
Ping @lal12 :-)

**Comment by lal12** [comment]
It indeed doesn't like function prototypes without name. This isn't totally trivially fixable. I've thought for a while about using a PEG grammar based parser instead.

**Comment by saghul** [comment]
So `int Fl_Window_x(Fl_Window *w);` should work, right?

**Comment by KaruroChori** [comment]
Yes it does, but it cannot count as a workaround if I have to do that hundreds of times :D.
This is the library I am using for reference: https://github.com/MoAlyousef/cfltk

I just got it to work and (almost) automatically translate all bindings for Bun ffi using https://github.com/Morglod/bun-ffi-gen and a lot of patchwork.
I was just hoping to do the same in txiki but failed for that and few more problems, like preprocessor macros not being expanded.

**Comment by lal12** [comment]
How about adding a bun ffi compatible wrapper?

**Comment by saghul** [comment]
> How about adding a bun ffi compatible wrapper?

That sounds like a good idea!

**Comment by KaruroChori** [comment]
As long as the dynamic library is open as RTLD_LAZY and not RTLD_NOW like Bun does :D.
Loading thousands of function signatures and enforcing them all to exist is really not ideal for fast loading time :/
Internally in txiki we are using uv_load, and if I recall correctly for unix systems it wraps dlopen with RTLD_LAZY. So if we keep this behaviour in place it should be fine.

**Comment by saghul** [comment]
Sounds good!

**Comment by saghul** [comment]
Given the lack of movement here, and since one can write the bindings by hand, I'm going to close this as as a known issue.
